mypy/dmypy_server.py

```diff
@@ -715,6 +716,29 @@ def refresh_file(module: str, path: str) -> list[str]:
 
         return messages
 
+    def _seen_and_ancestors(self, seen: set[str]) -> set[str]:
+        """Return the set of seen modules along with any ancestors not already in the set.
+
+        For example, given this set:
+
+            {"foo", "foo.bar", "a.b.c"}
+
+        ... we would expect this set to be returned:
+
+            {"foo", "foo.bar", "a.b.c", "a.b", "a"}
+
+        This is used to stop us from deleting ancestor modules from the graph
+        when their descendants have been seen.
+        """
+        seen_paths = seen.copy()
+        for module_path in seen:
+            while module_path := module_path.rpartition(".")[0]:
+                if module_path in seen_paths:
```

Comment:
I think we might be able to drop this conditional and just always add the module_path.

Reply:
It feels like it'd be quicker to skip this if we have a module with many submodules.